src/class_loader_core.cpp

```diff
@@ -545,6 +545,7 @@ void unloadLibrary(const std::string & library_path, ClassLoader * loader)
             ", keeping library %s open.",
             library_path.c_str());
         }
+        purgeGraveyardOfMetaobjects(library_path, loader, true);
```

Comment:
@ahcorde hmm, I think this was not here on purpose:

https://github.com/ros/class_loader/blob/7b03fe3acd442b6adaec8458408cb198fe49c89f/src/class_loader_core.cpp#L235-L242

But I wonder why https://github.com/ros/class_loader/blob/471c398d6b4d2a9936474aa08aee3691c89999f8/test/unique_ptr_test.cpp#L174 isn't failing for you though.

Reply:
What about this @ahcorde ?